crates/network-primitives/src/traits.rs

```diff
@@ -44,6 +45,22 @@ pub trait TransactionResponse {
     /// Input data
     #[doc(alias = "calldata")]
     fn input(&self) -> &Bytes;
+
+    /// Returns the gas price formatted for the RPC response.
+    fn gas_price(tx: &impl alloy_consensus::Transaction, base_fee: Option<u64>) -> u128;
+
+    /// Returns the max fee per gas.
+    fn max_fee_per_gas(tx: impl alloy_consensus::Transaction) -> Option<u128>;
```

Comment:
these are a bit weird, unsure why they need to be part of this trait

Reply:
so computation for l1 tx types can be reused by l2, much better than l2s doing copy paste